dockerfile_scripts/install_libnss_determined.sh

```diff
@@ -6,18 +6,24 @@ set -e
 # the container at runtime. This is critical for supporting non-root shell,
 # which in turn is critical for non-root distributed training.
 #
-# Also, to work around a behavior in OpenShift 4.+ [1], place `determined`
-# NSS plug-in to be searched first instead of last so Determined-defined
-# entries are used when there are conflicts with infrastructure-based
-# entries in these identity databases: passwd, shadow, group.
+# By default, the NSS plugin is placed last, which means it won't override any
+# settings present in the container's /etc/passwd.  However, if `--first` is
+# passed to the script, the NSS plugin is placed first.  This is necessary when
+# another system would conflict with Determined's system requirements.  An
+# exmaple is OpenShift 4.+ [1].
 #
 # [1] https://cloud.redhat.com/blog/a-guide-to-openshift-and-uids "By default,
 # OpenShift 4.x appends the effective UID into /etc/passwd of the Container
 # during the creation of the Pod."
 
-make -C /tmp/det_dockerfile_scripts/libnss_determined libnss_determined.so.2 install \
-    && sed -i -E '
+make -C /tmp/det_dockerfile_scripts/libnss_determined libnss_determined.so.2 install
+
+if [ "$1" = "--first" ] then
+    sed -i -E '
         /^(passwd|group|shadow):/{
             s/[ \t]determined//;
             s/:/: determined/;
         }' /etc/nsswitch.conf
+else
+    sed -E -i -e 's/^((passwd|shadow|group):.*)/\1 determined/' /etc/nsswitch.conf
```

Comment:
It bugs me that this one is `-E -i -e` and the other is `-i -E`, not that it should make a functional difference.

Reply:
removed second `-e`

Reply:
nvm, added the first `-e` instead to be more explicit.